### DISCUSSION TITLE
Windows Long File Names problem

### DISCUSSION THREAD

**Discussion by DzikiWojt** [sumatrapdfreader/sumatrapdf#4537] [Q&A]
Is it possible to enable Long file name (path) in Windows build ? 
I have a thousands of documents in folders longer than 256 long.
My endless effort to open documents in SumatraPDF is a pain...
Of course AcrobatReader do it without any problem or few other programs. 
Windows settings to enable Long File Names is of course enabled.

Maybe something in advanced settings SumatraPDF file should I enable ?

Normally few thousands of file I have on my local disk like:
C:\\some\folder\to\file\thefile.pdf

and dozens of thousands of file I have in VFS like:
\\some\remote\folder\with\files\thefile.pdf

I cannot open any of them if file with path is longer than 256 chars :-(

Sumatra 3.5.2 (64 bit Windows)

**Comment by kjk** [comment]
How are you opening them? File / Open? Drag & drop? Double-click in explorer?

What happens exactly?

Does it happen in [pre-release](https://www.sumatrapdfreader.org/prerelease) ?

Can you get and post [log](https://www.sumatrapdfreader.org/docs/Debugging-Sumatra#getting-logs) ?

**Reply by kjk** [reply]
I was able to pen file with full path longer than 256 in 3 different ways.

Windows just creates a shorter path.

So you would have to provide more information.

Also what is `Windows settings to enable Long File Names` ?

**Reply by DzikiWojt** [reply]
![Win10-Register](https://github.com/user-attachments/assets/4b6ce9b1-7eb0-46f8-9bac-041ef1b4a77b)

This is my Windows 10 registry with Long File Names enabled.

Example Path which do not open correctly:

\\wabnetdata\Media\[Ksiazki]\[Nowe]\[Zdrowie, Leczenie, Odzywianie]\[Medycyna Chinska]\Autor Nieznany - Huang Di nei jing su wen. Nature, knowledge, imagery in an ancient chinese medical text (tlumaczyl Paul U. Unschuld) [EN]\Huang_Di_Nei_Jing_Su_Wen-Paul_Unschuld.pdf

If I change file name from
Huang_Di_Nei_Jing_Su_Wen-Paul_Unschuld.pdf
to
a.pdf

then it will be full path shorter than 256 chars and then open correctly.

![SumatraPDF-3 5 2-CannotOpen](https://github.com/user-attachments/assets/5de2304c-1a1b-4868-8de8-0515b77b85ed)

In windows explorer or Double Commander double click on "pdf" file makes this error in SumatraPDF.
Also Right Click and from context menu choose open with SumatraPDF give the same error.

**Reply by DzikiWojt** [reply]
But this is very funny!!!!!

When I click in SumatraPDF 3.5.2 - File->Open and choose file manually - then it open !!!!
I downloaded new Sumatra yesterday, until yesterday I use a long time 3.2 version. And 3.2 version do not open from File->Open also, so I'm supprised, only from windows position 3.5.2 do not open file from long file path correctly.
I should mention ex. Adobe Acrobat Reader open long file path from explorer or double commander correctly.

**Comment by DzikiWojt** [comment]
[sumatra-log.txt](https://github.com/user-attachments/files/17083876/sumatra-log.txt)

This is my log when I tried to open file from windows explorer

**Comment by kjk** [comment]
I've created https://github.com/sumatrapdfreader/sumatrapdf/issues/4542 to track this.

**Comment by zolotovi** [comment]
Faced with the same issue when trying to open PDF file using double-click in explorer (where total path length > 260 symbols).
When support for long file names in Windows 10 is off SumatraPDF just hungs, otherwise when long file names in Windows 10 is on (according to https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry) SumatraPDF crashes.

**Comment by jw5801** [comment]
I can open files with longer path length with no difficulties either opening from explorer or from the file>open dialog.

But if I'm switching between files in a folder using ctrl+shift+left or ctrl+shift+right I get an error on any file with a path longer than the 260 character limit.

I'm using Sumatra 3.5.2 64-bit on Win11.

**Comment by kjk** [comment]
I've fixed some issues in https://www.sumatrapdfreader.org/prerelease 
If you still have issues in that build, please open a new issue with details
